tests/h/session_test.py

```diff
@@ -85,14 +141,106 @@ def test_profile_sorts_groups():
         FakeGroup('b', 'Group B'),
         FakeGroup('a', 'Group B'),
     ]
-    request = mock.Mock(authenticated_user=fake_user)
+    request = mock.Mock(authenticated_user=fake_user,
+                        auth_domain=fake_user.authority)
     profile = session.profile(request)
 
     ids = [group['id'] for group in profile['groups']]
 
     assert ids == ['__world__', 'c', 'a', 'b']
 
 
+def test_world_group_in_authenticated_profile(fake_user, auth_domain):
+    request = mock.Mock(authenticated_user=fake_user, auth_domain=auth_domain)
+
+    result = session.profile(request)
+
+    assert '__world__' in [g['id'] for g in result['groups']]
+
+
+def test_world_group_in_anonymous_profile(auth_domain):
+    request = mock.Mock(authenticated_user=None, auth_domain=auth_domain)
+
+    result = session.profile(request)
+
+    assert '__world__' in [g['id'] for g in result['groups']]
+
+
+def test_world_group_not_in_third_party_profile(third_party_user, auth_domain):
+    request = mock.Mock(authenticated_user=third_party_user,
+                        auth_domain=auth_domain)
+
+    result = session.profile(request)
+
+    assert '__world__' not in [g['id'] for g in result['groups']]
+
+
+def test_world_group_is_public_in_profile(fake_user, auth_domain):
+    request = mock.Mock(authenticated_user=fake_user, auth_domain=auth_domain)
+
+    profile = session.profile(request)
+    world_group = [g for g in profile['groups'] if g['id'] == '__world__'][0]
+
+    assert world_group['public'] is True
+
+
+def test_private_group_is_not_public_in_profile(auth_domain):
```

Comment:
That’s certainly an option, although in a [prior discussion with @chdorner](https://github.com/hypothesis/h/pull/4317/commits/5d386431d9f9f995299fab29d2ad9cbdcac728ce#r97614858) we came to the conclusion that duplicating the tests was a reasonable option, under the premise that we’re hoping to get rid of `session.model` in the future.